pkg/extract/redhat/csaf/csaf.go

```diff
@@ -680,216 +693,347 @@ func walkVulnerabilities(vulns []csaf.Vulnerability, pids []csaf.ProductID) (map
 				return rs
 			}(),
 			Published: utiltime.Parse([]string{"2006-01-02T15:04:05-07:00"}, v.ReleaseDate),
-		})
+		}
 	}
-	return vassm, vs, nil
+	return vassm, vm, nil
 }
 
-func buildDataComponents(baseAdvisory advisoryContentTypes.Content, baseVulnerabilities []vulnerabilityContentTypes.Content, pm map[csaf.ProductID][]product, vassm map[string]map[csaf.ProductID]ass) ([]advisoryTypes.Advisory, []vulnerabilityTypes.Vulnerability, []detectionTypes.Detection, error) {
-	vpvm := make(map[string]map[csaf.ProductID]map[string]vcTypes.Criterion)
-	for pid, ps := range pm {
-		for cve, assm := range vassm {
-			if vpvm[cve] == nil {
-				vpvm[cve] = make(map[csaf.ProductID]map[string]vcTypes.Criterion)
-			}
-			if vpvm[cve][pid] == nil {
-				vpvm[cve][pid] = make(map[string]vcTypes.Criterion)
-			}
-
-			for _, p := range ps {
-				vc, err := func() (*vcTypes.Criterion, error) {
-					switch assm[pid].status.product_status {
-					case "fixed":
-						if p.name == "" && p.version == "" {
-							return nil, nil
-						}
+func invertVassGroup(vassm map[string]map[csaf.ProductID]ass, pm map[csaf.ProductID][]product) (map[string][]csaf.ProductID, error) {
+	pidToVassGroup := make(map[csaf.ProductID]map[string]ass)
 
-						switch p.arch {
-						case "src":
-							return nil, nil
-						default:
-							return &vcTypes.Criterion{
-								Vulnerable: true,
-								FixStatus:  &fixstatusTypes.FixStatus{Class: fixstatusTypes.ClassFixed},
-								Package: vcPackageTypes.Package{
-									Type: vcPackageTypes.PackageTypeBinary,
-									Binary: &vcBinaryPackageTypes.Package{
-										Name: func() string {
-											if p.modularitylabel != "" {
-												return fmt.Sprintf("%s::%s", p.modularitylabel, p.name)
-											}
-											return p.name
-										}(),
-										Architectures: []string{p.arch},
-										Repositories:  p.repositories,
-									},
-								},
-								Affected: &affectedTypes.Affected{
-									Type:  rangeTypes.RangeTypeRPM,
-									Range: []rangeTypes.Range{{LessThan: p.version}},
-									Fixed: []string{p.version},
-								},
-							}, nil
-						}
-					case "affected":
-						if p.name == "" {
-							return nil, errors.New("name is empty")
-						}
+	for cveID, assm := range vassm {
+		for pid, ass1 := range assm {
```

Comment:
pass: product advisory severity status
```suggestion
		for pid, pass := range assm {
```